provider/datasource_sites.go

```diff
@@ -0,0 +1,213 @@
+						},
+						"block_duration_secs": {
+							Type:        schema.TypeInt,
+							Computed:    true,
+							Description: "Duration to block an IP in seconds",
+						},
+						"block_http_code": {
+							Type:        schema.TypeInt,
+							Computed:    true,
+							Description: "HTTP response code to send when when traffic is being blocked",
+						},
+						"block_redirect_url": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "URL to redirect to when blockHTTPCode is 301 or 302",
+						},
+						"created": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Created RFC3339 date time",
+						},
+						"display_name": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Display name of the site",
+						},
+						"events_uri": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Reference to the site's events",
+						},
+						"header_links_uri": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Reference to the site's header links",
+						},
+						"integrations_uri": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Reference to the site's integrations",
+						},
+						"members_uri": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Reference to the site's members",
+						},
+						"monitors_uri": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Reference to the site's monitors",
+						},
+						"name": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Identifying name of the site",
+						},
+						"redactions_uri": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Reference to the site's redactions",
+						},
+						"requests_uri": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Reference to the site's requests",
+						},
+						"suspicious_ips_uri": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Reference to the site's suspicious IPs",
+						},
+						"top_attacks_uri": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Reference to the site's top attacks",
+						},
+						"whitelist_uri": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Reference to the site's whitelist",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func readSites(d *schema.ResourceData, m any) error {
+	pm := m.(providerMetadata)
+	sc := pm.Client
+	corp := pm.Corp
+
+	// API documentation:
+	// https://docs.fastly.com/signalsciences/api/#_corps__corpName__sites_get
+	sites, err := sc.ListSites(corp)
+	if err != nil {
+		return err
+	}
+
+	d.SetId("list_sites")
```

Comment:
Previously I was dynamically generating a dynamic ID but opted for a static ID as I don't imagine there will be multiple instances of the `sigsci_sites` data source. If we _do_ think that is a possibility then we should make this dynamically generated (possibly using something like https://github.com/google/uuid to generate the ID).